website/docs/reference/opspec/index.md

```diff
@@ -101,5 +34,55 @@ Reference:
 - [socket](types/socket.md)
 - [string](types/string.md)
 
-## Scoping
-Variables in opspec are scoped to each operation. [Parameters](../op/parameter/index.md) allow passing values into or out of this scope.
\ No newline at end of file
+### Scoping
+
+A variable is created when defined as an [`input`](op.yml#inputs) or [`output`](op.yml#inputs), bound implicitly to an `output` when making an [op call](op.yml/call/op), or declared with [loop variables](./op.yml/loop-variable). A variable name is in scope from creation through the lifetime of it's `op.yml` file. Variable scope does not extend into [op calls](./op.yml/call/op).
+
+Variables can be referenced using the syntax `$(name)`.
+
+#### Scoping example
+
+```yml
```

Comment:
You might consider adding headings so these examples can be linked and so it's clear where one starts and the next begins

Reply:
Line 43 is the heading - this is all one example

Reply:
I personally can't follow it if that's the case; it all flows together and I lose track of what's happening.